Dockerfile

```diff
@@ -76,6 +77,7 @@ ENV TZ="GMT" \
     CUPS_IP="" \
     CUPS_ACCESS_LOGLEVEL="config" \
     # example: lpadmin -p Epson-RX520 -D 'my RX520' -m 'gutenprint.5.3://escp2-rx620/expert' -v smb://user:pass@host/Epson-RX520"
-    CUPS_LPADMIN_PRINTER1=""
+    CUPS_LPADMIN_PRINTER1="" \
+    PRE_INIT_HOOK="/root/${PRE_INIT_HOOK_NAME}"
```

Comment:
I would simplify the Dockerfile changes to only copy the default file `COPY pre-init-script.sh /root/` (and chmod +x)

everything else (incl. default path) I would only add to `start-cups.sh`

reason being: if someone builds their own image, they can add files (our mount them) wherever they want and only need to pass a new ENV to the script to be executed instead of the default. And there is no need to build your own image with an ARG to run a different script, if it's controlled only by the ENV.